### ISSUE TITLE
Probleme with microphones with some new samsung devices (2022-2023 modeles

### ISSUE THREAD

**Issue by saphir29000** [BearWare/TeamTalk5#1762]
Hello, with some new samsung devices, from friends, they have z-flip 4, s22 and from me, with my s23 ultra, we have only the left microphone working and others hears us on the left side with very low volume at 50%, and some others friends with their galaxy listed above, they don't have this issue.

**Comment by amirmahdifard** [comment]
how did you open thos that has no bug or any lable?

**Comment by valiant8086** [comment]
didn't notice if it is not stereo, but very low volume with my s21 even with mic level set to 100% (normally a very bad idea on my Pixel phone indeed). may be related?

**Comment by CoBC** [comment]
Oh, I'm able to reproduce this now, I'm not able previously, but it suddenly appeared. Very lowd volume and only left side. Very strange, no idea on how to resolve it.

**Comment by beqabeqa473** [comment]
@bear101 colud you please expose listing and setting of input devices from opensles to c-api if there is no such? i can't remember if they exist in c-api?
In that case we would be able to make a selector and let user to choose needed input source.

**Comment by bear101** [comment]
@beqabeqa473 On Android (and iOS) there's not really a list of source devices like on e.g. Windows and Linux. On Android I just create two device: a default device and a voice communication device. You can find the code for this [here](https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkLib/avstream/OpenSLESWrapper.cpp#L908).

On iOS I do something similar w.r.t. creating sound devices but from the UI application I'm able to change audio routing. Therefore iOS does not need a list of sound devices.

**Comment by beqabeqa473** [comment]
Yes, i understand that there are no list of devices, but on android there are input sources for camcorder, voice communication, raw unprocessed, voice call, default and others.
I meen that.

**Comment by shaghayegh1382** [comment]
@beqabeqa473 @bear101 @CoBC Same issue with samsung galaxy a73 here, microphone gain is 100% but my audio volume is still very low for others and it can be heard hardly. When its 50% it cannot be heard.

**Comment by Ramon-Salazar** [comment]
It appeared because the newer samsung phones don't support Recorder samplerate  of48KHZ, that's why, so by setting it to 24KHZ it fixed since newer samsung phones have a very high quality microphone

**Comment by bear101** [comment]
Why would a high quality microphone record at 24KHz? 48KHz would make more sense.

**Comment by hwangsihu** [comment]
I can't reproduce it now. Some of my friends seem to use that model, and it sounds really good.
